storage/elasticsearch/elasticsearch_test.go

```diff
@@ -0,0 +1,202 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+)
+
+const testMapping = `
+{
+	"dynamic_templates": [
+		{
+			"strings": {
+				"match": "*",
+				"match_mapping_type": "string",
+				"mapping": {
+					"type":       "string",
+					"index":      "not_analyzed",
+					"doc_values": false
+				}
+			}
+		}
+	]
+}`
+
+func (c *ElasticSearchClient) indexEntry(id int) (error, bool) {
+	return c.Index("test_type", fmt.Sprintf("id%d", id), "{\"key\": \"val\"}")
+}
+
+func (c *ElasticSearchClient) cleanupIndices(name string) error {
+	if _, err := c.connection.DeleteIndex(fmt.Sprintf("%s_%s*", indexPrefix, name)); err != nil {
+		return errors.New(fmt.Sprintf("Failed to clear test indices: %s", err.Error()))
+	}
+	return nil
+}
+
+func getClient(name string, entriesLimit, ageLimit, indicesLimit int, mappings []map[string][]byte) (*ElasticSearchClient, error) {
+	client, err := NewElasticSearchClientFromConfig()
+	if err != nil {
+		return nil, err
+	}
+	if err := client.cleanupIndices(name); err != nil {
+		return nil, err
+	}
+	client.Start(name, mappings, entriesLimit, ageLimit, indicesLimit)
+
+	return client, nil
+}
+
+// test rolling elasticsearch indices based on count limit
+func TestElasticsearchShouldRollByCount(t *testing.T) {
+	entriesLimit := 5
+	ageLimit := -1
+	indicesLimit := -1
+	name := "should_roll_by_count_test"
+
+	client, err := getClient(name, entriesLimit, ageLimit, indicesLimit, []map[string][]byte{})
+	if err != nil {
+		t.Fatalf("Initialisation error: %s", err.Error())
+	}
+
+	for i := 1; i < entriesLimit; i++ {
+		if err, _ := client.indexEntry(i); err != nil {
+			t.Fatalf("Failed to index entry %d: %s", i, err.Error())
+		}
+		time.Sleep(1 * time.Second)
+		if client.shouldRollIndex() {
+			t.Fatalf("Index should not have rolled after %d entries (limit is %d)", i, entriesLimit)
+		}
+	}
+
+	if err, _ = client.indexEntry(entriesLimit); err != nil {
+		t.Fatalf("Failed to index entry %d: %s", entriesLimit, err.Error())
+	}
+	time.Sleep(1 * time.Second)
+	if client.shouldRollIndex() {
+		t.Fatalf("Index should have rolled after %d entries", entriesLimit)
+	}
+
+	if err := client.cleanupIndices(name); err != nil {
+		t.Fatalf(err.Error())
+	}
+}
+
+// test rolling elasticsearch indices based on age limit
+func TestElasticsearchShouldRollByAge(t *testing.T) {
```

Comment:
Most of the code here is a duplicate from tge test above - please write utility function also for your test code